### DISCUSSION TITLE
Fluent interface for caching Eloquent queries

### DISCUSSION THREAD

**Discussion by th-alanreed** [laravel/framework#47128] [Ideas]
It would be nice if you could do something like this:

`$count = User::cache(now()->addHour())->count();`

This is a simple query but you can imagine a more complicated query you would want to cache. The cache key could the SQL query plus the bindings.

That is a lot cleaner than writing this:

    $count = Cache::remember('user_count', now()->addHour(), function() {
        return User::count();
    });

**Comment by henzeb** [comment]
You should take a look at. https://leqc.renoki.org/cache-tags/query-caching

**Reply by th-alanreed** [reply]
Thanks
